wazo_phoned/plugins/yealink/services.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019-2020 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from requests.exceptions import RequestException
+from wazo_phoned.plugin_helpers.client.exceptions import NoSuchUser
+
+
+class YealinkService:
+    def __init__(self, amid_client, confd_client):
+        self.amid = amid_client
+        self.confd = confd_client
+
+    def update_dnd(self, user_uuid, enabled):
+        try:
+            self.confd.users(user_uuid).update_service('dnd', {'enabled': enabled})
+        except RequestException as e:
+            response = getattr(e, 'response', None)
+            status_code = getattr(response, 'status_code', None)
+            if status_code == 404:
+                raise NoSuchUser(user_uuid)
+            raise
+
+    def notify_dnd(self, user_uuid, enabled):
+        lines = self._find_lines(user_uuid)
+        for line in lines:
+            endpoint = line.get('name')
+            if endpoint:
```

Comment:
We should verify if the line has `endpoint_sip` to avoid sending PJSIPNotify with sccp/custom lines